poppy/wfe.py

```diff
@@ -323,3 +324,32 @@ def get_opd(self, wave, units='meters'):
         if units == 'waves':
             opd /= wave.wavelength.to(u.meter).value
         return opd
+
+
+class StatsticalPSDWFE(WavefrontError):
+    """
+    Statistical PSF WFE class from power law for optical noise.
+
+    Input WFE in radians RMS.
+    """
+
+    @utils.quantity_input(wfe=u.rad)
+    def __init__(self, name='PSD WFE', index=3.0, wfe=0.25, **kwargs):
+
+        super(WavefrontError, self).__init__(name=name, **kwargs)
+        self.index = index
+        self.wfe_rad = wfe
+
+    def get_opd(self, wave, units='meters'):
+
+        y, x = self.get_coordinates(wave)
+        rho, theta = _wave_y_x_to_rho_theta(y, x, self.radius.to(u.meter).value)
+        psd = np.power(rho, -self.index)
```

Comment:
I need to add setting this to zero at the center for piston removal.

Reply:
@mperrin Is it really necessary here to explicitly set the radial array to zero in the center? With an odd-pixeled wave object, the function _wave_y_x_to_rho_theta() takes care of that anyway as far as I can see, and with an even-pixeled array there is no center pixel.